src/plonk/circuit.rs

```diff
@@ -6,21 +6,27 @@ use super::Error;
 use crate::arithmetic::Field;
 
 use crate::poly::Rotation;
-/// This represents a wire which has a fixed (permanent) value
+
+/// Enum over wires with variable values
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-pub struct FixedWire(pub usize);
+pub enum Wire {
+    /// A wire which has a witness-specific value
+    AdviceWire(usize),
+    /// A wire which has an externally specified value
+    AuxWire(usize),
+}
 
-/// This represents a wire which has a witness-specific value
+/// This represents a wire which has a fixed (permanent) value
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-pub struct AdviceWire(pub usize);
+pub struct FixedWire(pub usize);
 
 /// This trait allows a [`Circuit`] to direct some backend to assign a witness
 /// for a constraint system.
 pub trait Assignment<F: Field> {
     /// Assign an advice wire value (witness)
     fn assign_advice(
         &mut self,
-        wire: AdviceWire,
+        wire: Wire,
```

Comment:
This is an example of why we shouldn't consolidate these types. It would be possible to call `assign_advice` on an aux wire.